### ISSUE TITLE
关于learn过程中Nan问题

### ISSUE THREAD

**Issue by zhouchunyi** [MorvanZhou/Reinforcement-learning-with-tensorflow#99]
Morvan老师，我最近开始学习强化学习，根据我的项目需求写了一个env，用policy_gradient方法（是REINFORCE算法吧^_^）跑的时候，每次learn后，prob_weights输出都是[nan,..,nan]。刚上手，想请问这个问题该如何解决？看网上好像说是在求loss时log计算部分造成的？您的源码是下面形式：
`neg_log_prob = tf.reduce_sum(-tf.log(self.all_act_prob) * tf.one_hot(self.tf_acts, self.n_actions), axis=1)`
`loss = tf.reduce_mean(neg_log_prob * self.tf_vt)`

**Comment by MorvanZhou** [comment]
出现nan最主要的原因是数值过大，请检查你的欢迎是否输入端，或者reward定义过大或过小？

**Comment by zhouchunyi** [comment]
Morvan老师，感谢您的回复，我重新检查后，发现nan出现在每回合normalize reward中，
`discounted_ep_rs /= np.std(discounted_ep_rs) `，请问这是否也是输入造成的？
另外，想问一下DQN算法中当数据量比较大的时候，memory_size是否也要相应扩大些，对应文章的batch_size设为100。

**Comment by MorvanZhou** [comment]
如果在rewards的变动幅度不是很大，那么也可以不除以std，除以std是为了减小reward的variance，让网络可以更好的学习，gradient不会突然变太大。
第二个问题，memory size 扩大肯定是有帮助的。

**Comment by zhouchunyi** [comment]
谢谢老师的及时回复！非常受用！
